gitlab/resource_gitlab_project.go

```diff
@@ -301,6 +317,22 @@ func resourceGitlabProjectCreate(d *schema.ResourceData, meta interface{}) error
 		options.ImportURL = gitlab.String(v.(string))
 	}
 
+	if v, ok := d.GetOk("template_name"); ok {
+		options.TemplateName = gitlab.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("template_project_id"); ok {
+		options.TemplateProjectID = gitlab.Int(v.(int))
```

Comment:
Same thing here with `options.TemplateProjectID`, should it be declared somewhere? Also bit of a warning: I tried opening up my own PR based on yours here: https://github.com/gitlabhq/terraform-provider-gitlab/pull/377 but when I had both `TemplateName` and `TemplateProjectID` in `CreateProjectOptions`, I kept getting an error that TemplateName and TemplateProjectID had to be mutually exclusive. Do you know how to fix that?

Reply:
I struggled with the mutually exclusive error for a while, if you figure it out please let me know how! If not, perhaps we can stick to either TemplateName or TemplateProjectID but not both- since the docs say TemplateProjectID is preferable, maybe just that? Ideally we get around the exclusivity error though.

Reply:
I already see in some codes a way to achieve this (probably in ValidateFunc), will check if possible otherwise I will update the doc (Gitlab doc doesn't said something like that but it's 'trivial')

Reply:
Ok found an example here, it's tested in code directly https://github.com/terraform-providers/terraform-provider-azurerm/blob/687e20310ee035694a83606a0815e702dd357f12/azurerm/internal/services/batch/batch_pool_resource.go#L542
So I added the same test and in acceptance testing that the project doesn't exists.